app/scripts/controllers/login-v2-controller.js

```diff
@@ -58,34 +58,50 @@ angular.module('stellarClient').controller('LoginV2Ctrl', function($scope, $stat
     }
 
     // We don't have to run $scope.$apply because it's wrapped in singletonPromise
-    return StellarWallet.getWallet(params).then(function(wallet) {
-      if ($scope.rememberMe) {
-        session.rememberUser();
-      }
-
-      session.login(new Wallet({
-        version: 2,
-        id: wallet.getWalletId(),
-        key: wallet.getWalletKey(),
-        keychainData: wallet.getKeychainData(),
-        mainData: wallet.getMainData(),
-        walletV2: wallet
-      }));
-      $state.go('dashboard');
-    }).catch(StellarWallet.errors.TotpCodeRequired, function() {
-      $scope.loginError = "2-Factor-Authentication code is required to login.";
-    }).catch(StellarWallet.errors.ConnectionError, function() {
-      $scope.loginError = "Error connecting wallet server. Please try again later.";
-    }).catch(function(e) {
-      if (e.name && e.name === 'Forbidden') {
-        return $q.reject(e);
-      }
-      Raven.captureMessage('StellarWallet.getWallet unknown error', {
-        extra: {
-          error: e
+    return StellarWallet.getWallet(params)
+      .tap(function(wallet) {
+        if ($scope.rememberMe) {
+          session.rememberUser();
+        }
+        session.login(new Wallet({
+          version: 2,
+          id: wallet.getWalletId(),
+          key: wallet.getWalletKey(),
+          keychainData: wallet.getKeychainData(),
+          mainData: wallet.getMainData(),
+          walletV2: wallet
+        }));
+      })
+      .then(function(wallet) {
+        var bugDeploy   = new Date('2014-11-17');
```

Comment:
I would prefer that we extract this into it's own method, with a jsdoc block that further explains the issue we are handling.

This specific scenario's details is something that will be lost to time if we don't document it.